### PR TITLE
Add Avro Reader options classes to pylibcudf

### PR DESCRIPTION
**Pull request by Matt711**
## Description

Apart of #17565
## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/pylibcudf/pylibcudf/io/avro.pyi [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
 
-__all__ = ["read_avro"]
-
-def read_avro(
-    source_info: SourceInfo,
-    columns: list[str] | None = None,
-    skip_rows: int = 0,
-    num_rows: int = -1,
-) -> TableWithMetadata: ...
+__all__ = ["AvroReaderOptions", "AvroReaderOptionsBuilder", "read_avro"]
```
Comment (Matt711) [inline python/pylibcudf/pylibcudf/io/avro.pyi:4]: I don't see `__all__`'s in other type stub files. Should it be included?

File: python/pylibcudf/pylibcudf/io/avro.pyx [hunk 2]
```diff
@@ -10,52 +10,120 @@ from pylibcudf.libcudf.io.avro cimport (
...
+        Parameters
+        ----------
+        col_names : list
+            List of column names
+
+        Returns
+        -------
+        AvroReaderOptionsBuilder
+        """
+        cdef vector[string] vec
```
Comment (mroeschke) [inline python/pylibcudf/pylibcudf/io/avro.pyx:60]: So if `col_names` is None or length 0, the `self.c_obj.columns(vec)` will work correctly here?

File: python/pylibcudf/pylibcudf/io/avro.pyi (hunk 1 above)
Comment (mroeschke) [reply python/pylibcudf/pylibcudf/io/avro.pyi:4]: IMO yes, ideally the stubs should also have `__all__`

File: python/pylibcudf/pylibcudf/io/avro.pyx (hunk 2 above)
Comment (Matt711) [reply python/pylibcudf/pylibcudf/io/avro.pyx:60]: Yes for length 0 (there's a[ test case](https://github.com/rapidsai/cudf/pull/17599/files#diff-c9ec7da97b86cf64b800e6821ef4ea137e119c13f08af8f95b9037fb8b3736beR65) for that). I'll add a test case for `None` too.

File: python/pylibcudf/pylibcudf/io/avro.pyi (hunk 1 above)
Comment (Matt711) [reply python/pylibcudf/pylibcudf/io/avro.pyi:4]: Ok, will follow up to add `__all__`

File: python/pylibcudf/pylibcudf/io/avro.pyx (hunk 2 above)
Comment (Matt711) [reply python/pylibcudf/pylibcudf/io/avro.pyx:79]: I used `set_columns` instead and added the none-check at the caller site.

**Review by mroeschke — APPROVED** [review]

**Pull request merged**